### PR TITLE
15711 legal name model updates

### PR DESCRIPTION
**Pull request by argush3**
*Issue #:* /bcgov/entity#15711

Note: these changes will be merged into **development_legal_name**, not **main**.  Until we have tested all the upcoming legal name work thoroughly against **development_legal_name**, we won't be merging **development_legal_name** into **main**.

*Description of changes:*
This PR was intended mainly to update the database model via alembic scripts and the SQLAlchemy models.  But since the tables involved were core to everything in the BE, a lot of code needed to be updated for the legal api to continue functioning.   

This was a first pass as it relates to updating the legal api code to get it in a state that passes the existing tests against the new db model.  More work will be done in the near future to the legal api as well as other BE services/jobs after data migration scripts have been created/run.

Note that some code and tables such will be removed later when the migration to Prod has taken place.

As there are a lot of changes in this PR, to help reviewers, the "primary changes" listed below are probably the files of most interest.

* **Primary changes**
  * alembic script update - legal-api/migrations/versions/99506f8f927c_legal_name_updates.py
  * model updates
    * legal-api/src/legal_api/models/alternate_name.py
    * legal-api/src/legal_api/models/colin_entity.py
    * legal-api/src/legal_api/models/entity_role.py
    * legal-api/src/legal_api/models/legal_entity.py
    * legal-api/src/legal_api/models/role_address.py
  * model tests
    * legal-api/tests/unit/models/test_alternate_name.py
    * legal-api/tests/unit/models/test_business.py
    * legal-api/tests/unit/models/test_colin_entity.py
    * legal-api/tests/unit/models/test_entity_role.py
    * legal-api/tests/unit/models/test_role_address.py
  * Existing Party/PartyRole model related logic needed was updated to use EntityRole/LegalEntity models.
* **Secondary changes**
  * A good chunk of the remaining files updated were mainly updating references to use the new model names and new db structure. 

Summary of model changes.  Note that this diagrams documents the key model changes but is not exhaustive.
image

Complete list of updates to database model.  Essentially this is a summary of what the alembic script(legal-api/migrations/versions/99506f8f927c_legal_name_updates.py) does.

image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by argush3** [comment]
@vysakh-menon-aot @kzdev420 @ChrisGabel23 please take a look when you have a minute.  I have put in a bunch of details in the PR description to hopefully help with reviewing this giant PR.  

I am still working on fixing the code smells that I can but I think it's in a state that can be reviewed.

**Comment by argush3** [comment]
@vysakh-menon-aot @kzdev420 @ChrisGabel23 Update on code smells.  I have fixed 16 code smells with 13 remaining now.  

I was hoping to leave the remaining code smells as is.  The reason being that I am changing a ton of code as is.  I don't want to increase the potential for new bugs to fix mainly pre-existing code complexity issues.

**Comment by ChrisGabel23** [comment]
@argush3 Could you put a pylint comment to disable the code complexity smell checks in the flagged code?

**Comment by argush3** [comment]
> @argush3 Could you put a pylint comment to disable the code complexity smell checks in the flagged code?

I don't think so.  I think someone with the right access levels can do it in sonarcloud can do it.  I'll see if I can get access.

**Comment by argush3** [comment]
Turns out my GH account didn't belong to BCGov Org so I couldn't do any non read-only activities in sonarcloud.  I was resolved the two TODO code smells.

File: legal-api/src/legal_api/models/legal_entity.py [hunk 1]
```diff
@@ -0,0 +1,743 @@
...
+            'tax_id',
+            'naics_key',
+            'naics_code',
+            'naics_description',
+            'start_date',
+            'jurisdiction',
+            'foreignIdentifier',
+            'foreignLegalName',
+            'foreignLegalType',
+            'foreignIncorporationDate',
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/models/legal_entity.py:204]: I fixed these by changing to snake case

Comment (argush3) [reply legal-api/src/legal_api/models/legal_entity.py:204]: ok, will update that.

**Review by vysakh-menon-aot — APPROVED** [review]

**Review by kzdev420 — APPROVED** [review]

**Pull request merged**